packages/docs-page/server/index.ts

```diff
@@ -1,11 +1,145 @@
+
+interface BaseOpts {
+  fallback?: GetStaticPathsResult['fallback']
+  revalidate?: number
+  product: string
+}
+
+export function getStaticGenerationFunctions(
+  opts:
+    | ({
+        basePath?: string
+        strategy: 'remote'
+      } & BaseOpts)
+    | ({
+        localContentDir: string
+        navDataFile: string
+        strategy: 'fs'
+      } & BaseOpts)
+): {
+  getStaticPaths: GetStaticPaths
+  getStaticProps: GetStaticProps
+} {
+  let loader: DataLoader
+
+  switch (opts.strategy) {
+    default:
+    case 'fs': {
+      const { strategy, ...restOpts } = opts
+      loader = new FileSystemLoader({ ...restOpts })
+      break
+    }
+    case 'remote': {
+      const { strategy, ...restOpts } = opts
+
+      loader = new RemoteContentLoader({
+        // derive basePath from __dirname, example: .next/server/pages/waypoint/docs
+        basePath: __dirname.split('/').slice(-1)[0],
+        ...restOpts,
+      })
+    }
+  }
+
+  return {
+    getStaticPaths: async (ctx) => {
+      const paths = await loader.loadStaticPaths(ctx)
+      return {
+        fallback: opts.fallback ?? 'blocking',
+        paths,
+      }
+    },
+    getStaticProps: async (ctx) => {
+      const props = await loader.loadStaticProps(ctx)
+      return {
+        props,
+        revalidate: opts.revalidate,
+      }
+    },
+  }
+}
+
+export interface GenerateStaticPathsContext {
+  /** @example 'data/docs-nav-data.json' */
+  navDataFile: string
+  /** @example 'content/docs' */
+  localContentDir: string
+  /**
+   * @default 'page'
+   */
+  paramId?: string
+  /**
+   * @example { name: 'Waypoint', slug: 'waypoint' }
+   */
+  product?: { name: string; slug: string }
+  /** @example 'docs' */
+  basePath?: string
+}
+
+/**
+ * @deprecated Use getStaticGenerationFunctions instead
+ */
+export function generateStaticPaths({
+  navDataFile,
+  localContentDir,
+  paramId,
+  product,
+  basePath,
+}: GenerateStaticPathsContext) {
+  const loader = new FileSystemLoader({
+    navDataFile,
+    localContentDir,
+    product: product.slug,
+    paramId,
+  })
+
+  return loader.loadStaticPaths()
+}
+
+export interface GenerateStaticPropsContext {
+  navDataFile: string
+  localContentDir: string
+  params: Record<string, string[]> // {} | { page: ["destroy"] }
+  product: { name: string; slug: string }
+  mainBranch?: string // = 'main',
+  remarkPlugins?: any[]
+  scope?: any // optional, I think?
+  paramId?: string
+  basePath: string // 'docs'
+}
+
+/**
+ * @deprecated Use getStaticGenerationFunctions instead
+ */
+export function generateStaticProps({
+  navDataFile,
+  localContentDir,
+  paramId,
+  product,
+  params,
+  remarkPlugins,
+  scope,
+  mainBranch,
+}: GenerateStaticPropsContext) {
+  const loader = new FileSystemLoader({
+    navDataFile,
+    localContentDir,
+    product: product.slug,
+    paramId,
+  })
+
+  return loader.loadStaticProps({ params, remarkPlugins, scope, mainBranch })
+}
```

Comment:
Avoid the breaking change, but mark as deprecated. Uses `FileLoader` only, we'll update the versioned docs stuff to use the new `getStaticGenerationFunctions`.